engine/src/test/java/io/zeebe/engine/processor/StreamProcessorTest.java

```diff
@@ -578,16 +587,25 @@ public void onRecovered(final ReadonlyProcessingContext context) {
   }
 
   @Test
-  public void shouldCreateSnapshotsEvenIfNoProcessorProcessEvent() {
+  public void shouldCreateSnapshotsEvenIfNoProcessorProcessEvent() throws InterruptedException {
```

Comment:
I did not understand the intention of this test. The name says no processor process event, but there is a processor processing the event. :thinking:

Reply:
The name is misleading, true; what it means is that between an open/close of the processor, even if we didn't process an event, on close we still snapshot.

You can see we process the event then close, then re-open the processor. This is because we don't take snapshots if we have never processed anything (i.e. if the last processed position is -1) - to support "empty" snapshots (e.g. index -1), we'd need to refactor Atomix more, so skipping at -1 seemed simpler. We can challenge that.

In the mean time, we can find a better name, any ideas?

Reply:
It sounds like `shouldNotCreateSnapshotIfNoProcessorProcessEvent()`.

Reply:
But we want to test that it does create one :sweat_smile: Let's just quickly clear it up on Zoom